rocketpy/Flight.py

```diff
@@ -2091,6 +2097,114 @@ def postProcess(self, interpolation="spline", extrapolation="natural"):
+                    3.14159265359 / 2 + np.arctan(abs(self.y[i][1]) / abs(self.x[i][1]))
+                )
+                continue
+            elif self.x[i][1] * self.y[i][1] > 0 and self.x[i][1] < 0:
+                bearing.append(
+                    3.14159265359 + np.arctan(abs(self.x[i][1]) / abs(self.y[i][1]))
+                )
+                continue
+            elif self.x[i][1] * self.y[i][1] < 0 and self.x[i][1] < 0:
+                bearing.append(
+                    3 * 3.14159265359 / 2
+                    + np.arctan(abs(self.y[i][1]) / abs(self.x[i][1]))
+                )
+                continue
+
+        # Store values of distance and bearing using approriate units
+        # self.distance = distance      # Must be in meters
+        # self.bearing = bearing        # Must be in radians
+
+        lat1 = np.pi * self.env.lat / 180  # Launch lat point converted to radians
+        lon1 = np.pi * self.env.lon / 180  # Launch long point converted to radians
+
+        R = self.env.earthRadius
+        lat2 = []
+        lon2 = []
+        # Applies the haversine equation to find final lat/lon coordinates
+        for i in range(len(distance)):
+            # Please notice that for distances lower than 1 centimeter the difference on latitude or longitude too small
+            if abs(self.x[i][1]) < 1e-2 and abs(self.y[i][1]) < 1e-2:
+                lat2.append(self.env.lat)
+                lon2.append(self.env.lon)
+                continue
+            elif abs(self.x[i][1]) < 1e-2:
+                lat2.append(
+                    (180 / np.pi)
+                    * np.arcsin(
+                        np.sin(self.env.lat) * np.cos(distance[i] / R)
+                        + np.cos(lat1) * np.sin(distance[i] / R) * np.cos(bearing[i])
+                    )
+                )
+                lon2.append(self.env.lon)
+                continue
+            elif abs(self.y[i][1]) < 1e-2:
+                lat2.append(self.env.lat)
+                lon2.append(
+                    (180 / np.pi) * lon1
+                    + np.arctan2(
+                        np.sin(bearing[i]) * np.sin(distance[i] / R) * np.cos(lat1),
+                        np.cos(distance[i] / R) - np.sin(lat1) * np.sin(lat2[i]),
+                    )
+                )
+                continue
+            else:
+                lat2.append(
+                    (180 / np.pi)
+                    * np.arcsin(
+                        np.sin(lat1) * np.cos(distance[i] / R)
+                        + np.cos(lat1) * np.sin(distance[i] / R) * np.cos(bearing[i])
+                    )
+                )
+                lon2.append(
+                    (180 / np.pi) * lon1
+                    + np.arctan2(
+                        np.sin(bearing[i]) * np.sin(distance[i] / R) * np.cos(lat1),
+                        np.cos(distance[i] / R) - np.sin(lat1) * np.sin(lat2[i]),
+                    )
+                )
```

Comment:
If I understood this correctly it might be nicer (and maybe faster) to do something like this:

```suggestion
        # Applies the haversine equation to find final lat/lon coordinates
        for i in range(len(distance)):
            # Please notice that for distances lower than 1 centimeter the difference on latitude or longitude too small
            if abs(self.x[i][1]) < 1e-2:
                lon2.append(self.env.lon)
            else:
                lon2.append(
                    (180 / np.pi) * lon1
                    + np.arctan2(
                        np.sin(bearing[i]) * np.sin(distance[i] / R) * np.cos(lat1),
                        np.cos(distance[i] / R) - np.sin(lat1) * np.sin(lat2[i]),
                    )
                )
            if abs(self.y[i][1]) < 1e-2:
                lat2.append(self.env.lat)
            else:
                lat2.append(
                    (180 / np.pi)
                    * np.arcsin(
                        np.sin(lat1) * np.cos(distance[i] / R)
                        + np.cos(lat1) * np.sin(distance[i] / R) * np.cos(bearing[i])
                    )
                )
```